### PR TITLE
fix(flow): optimize flow submitter about exception handler

### PR DESCRIPTION
**Pull request by krihy**
#### What this PR does / why we need it:
optimize `FlowTaskSubmitter` about exception handler,  if `BaseRuntimeFlowableDelegate` throw exception, we check the `ErrorEventDefinition` boundary with the node,  if the boundary error code match with current exception,  flowTaskCallBackApprovalService will approve the callback task with FAILED,  otherwise approve with COMPLETED, 
by  this pr flowTaskCallBackApprovalService do not care about the task type.  and flowNode status update by `ExecutionListener` boundary by the `ErrorEventDefinition`
 
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

this pr has test passed:
case1: when submit async task and cancel it , the flow status will be `CANCELED`
case2: when submit a invalid sql async task, the flow status will be `FAILED`
case3: when submit a normal sql async task,  the flow status will be `COMPLETED`

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yhilmare** [comment]
👏 great job

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskResultContextHolder.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @author yaobin
+ * @date 2024-05-17
+ * @since 4.2.4
+ */
+public class FlowTaskResultContextHolder {
+
+
+    public static final ThreadLocal<Map<String, Object>> RESULT_CONTEXT = new ThreadLocal<>();
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskResultContextHolder.java:28]: may private ?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java [hunk 2]
```diff
@@ -48,32 +60,102 @@ public class FlowTaskSubmitter implements JavaDelegate {
...
     }
 
-    private BaseRuntimeFlowableDelegate<?> getDelegateInstance(DelegateExecution execution) throws Exception {
-        Optional<FlowTaskInstance> flowTaskInstance = flowableAdaptor.getTaskInstanceByActivityId(
-                execution.getCurrentActivityId(), FlowTaskUtil.getFlowInstanceId(execution));
-        PreConditions.validExists(ResourceType.ODC_FLOW_TASK_INSTANCE, "activityId",
-                execution.getCurrentActivityId(), flowTaskInstance::isPresent);
+    private void handleException(DelegateExecution execution, FlowTaskInstance flowTaskInstance, Exception e,
+            List<FlowableBoundaryEvent<ErrorEventDefinition>> defs) {
+        String ids = execution.getProcessDefinitionId();
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java:95]: ids->processInstanceId

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java [hunk 3]
```diff
@@ -48,32 +60,102 @@ public class FlowTaskSubmitter implements JavaDelegate {
...
+        String aId = execution.getCurrentActivityId();
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java:96]: aId -> activityId

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskSubmitter.java]: okey

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskResultContextHolder.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/FlowTaskResultContextHolder.java]: yes,  done

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**